packages/utils/node-resolver-core/src/builtins.js

```diff
@@ -4,35 +4,39 @@ import {builtinModules} from 'module';
 
 export const empty: string = require.resolve('./_empty.js');
 
-// $FlowFixMe
-let builtins: {[string]: any, ...} = Object.create(null);
+let builtins: {[string]: {|name: string, range: ?string|}, ...} =
+  // $FlowFixMe
+  Object.create(null);
+
 // use definite (current) list of Node builtins
 for (let key of builtinModules) {
-  builtins[key] = empty;
+  builtins[key] = {name: empty, range: null};
 }
 
-builtins.assert = 'assert/';
-builtins.buffer = 'buffer/';
-builtins.console = 'console-browserify';
-builtins.constants = 'constants-browserify';
-builtins.crypto = 'crypto-browserify';
-builtins.domain = 'domain-browser';
-builtins.events = 'events/';
-builtins.http = 'stream-http';
-builtins.https = 'https-browserify';
-builtins.os = 'os-browserify/browser.js';
-builtins.path = 'path-browserify';
-builtins.process = 'process/browser.js';
-builtins.punycode = 'punycode/';
-builtins.querystring = 'querystring-es3/';
-builtins.stream = 'stream-browserify';
-builtins.string_decoder = 'string_decoder/';
-builtins.sys = 'util/util.js';
-builtins.timers = 'timers-browserify';
-builtins.tty = 'tty-browserify';
-builtins.url = 'url/';
-builtins.util = 'util/util.js';
-builtins.vm = 'vm-browserify';
-builtins.zlib = 'browserify-zlib';
+// These are mostly the versions considered "new enough" by
+// https://www.npmjs.com/package/node-libs-browser (with a few version bumps)
+builtins.assert = {name: 'assert/', range: '^2.0.0'};
+builtins.buffer = {name: 'buffer/', range: '^5.5.0'};
+builtins.console = {name: 'console-browserify', range: '^1.2.0'};
+builtins.constants = {name: 'constants-browserify', range: '^1.0.0'};
+builtins.crypto = {name: 'crypto-browserify', range: '^3.12.0'};
+builtins.domain = {name: 'domain-browser', range: '^3.5.0'};
+builtins.events = {name: 'events/', range: '^3.1.0'};
+builtins.http = {name: 'stream-http', range: '^3.1.0'};
+builtins.https = {name: 'https-browserify', range: '^1.0.0'};
+builtins.os = {name: 'os-browserify/browser.js', range: '^0.3.0'};
+builtins.path = {name: 'path-browserify', range: '^1.0.0'};
+builtins.process = {name: 'process/browser.js', range: '^0.11.10'};
+builtins.punycode = {name: 'punycode/', range: '^1.4.1'};
+builtins.querystring = {name: 'querystring-es3/', range: '^0.2.1'};
+builtins.stream = {name: 'stream-browserify', range: '^3.0.0'};
+builtins.string_decoder = {name: 'string_decoder/', range: '^1.3.0'};
+builtins.sys = {name: 'util/util.js', range: '^0.12.3'};
+builtins.timers = {name: 'timers-browserify', range: '^2.0.11'};
+builtins.tty = {name: 'tty-browserify', range: '^0.0.1'};
+builtins.url = {name: 'url/', range: '^0.11.0'};
+builtins.util = {name: 'util/util.js', range: '^0.12.3'};
+builtins.vm = {name: 'vm-browserify', range: '^1.1.2'};
+builtins.zlib = {name: 'browserify-zlib', range: '^0.2.0'};
```

Comment:
Should we load these dependencies from the `devDependencies` in package.json so they are always in sync?